### PR TITLE
Throw event when changing session id

### PR DESCRIPTION
**Pull request by wouterverstuyf**
## Type
- Enhancement

## Resolves the following issues
When a public profile login/logout the session id changes.
Could be handy for other modules to hook in through an event, to get the new Session ID.

## Pull request description
You can create a listener in your own module that detects the login/logout action. In the listener you can get the new session ID and do your stuff

/Frontend/Modules/MyModule/EventListener/ProfilesSessionIdChangedListener.php

<pre>
namespace Frontend\Modules\MyModule\EventListener;

use Common\Events\ForkSessionIdChangedEvent;

class ForkSessionIdChangedListener
{
    public function onForkSessionIdChanged(ForkSessionIdChangedEvent $event): void
    {
        // do your stuff
        // $event->getSessionId()
        // $event->getOldSessionId()
    }
}
</pre>

### CODE REVIEW

**Comment by wouterverstuyf** [comment]
Question : when you login/logout in the backend section, the Session ID also changes. Should this be event be triggered from the backend login/logout function also? Is it ok to trigger a frontend event from within the backend section?

**Comment by carakas** [comment]
@wouterverstuyf I would add it in the Common namespace, not in the profiles one. It is tied to the session, not a profile

File: src/Frontend/Modules/Profiles/Engine/Authentication.php [hunk 1]
```diff
@@ -243,13 +247,21 @@ public static function login(int $profileId, bool $remember = false): void
...
+        );
+
         // load the profile object
         self::$profile = new FrontendProfilesProfile($profileId);
     }
 
     public static function logout(): void
     {
         $session = FrontendModel::getSession();
+        $oldSession = FrontendModel::getSession()->getId();
```
Comment (carakas) [inline src/Frontend/Modules/Profiles/Engine/Authentication.php:263]: you can use $session->getId() here

**Comment by wouterverstuyf** [comment]
> @wouterverstuyf I would add it in the Common namespace, not in the profiles one. It is tied to the session, not a profile, and then you can throw the event in the backend as well

Yes, got the same feeling this morning, I check it ;)

File: src/Frontend/Modules/Profiles/Event/ProfilesSessionIdChangedEvent.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    /**
+     * @var string
+     */
+    protected $sessionId;
+
+    public function __construct(string $oldSessionId)
+    {
+        $this->oldSessionId = $oldSessionId;
+        $this->sessionId = FrontendModel::getSession()->getId();
```
Comment (carakas) [inline src/Frontend/Modules/Profiles/Event/ProfilesSessionIdChangedEvent.php:23]: Pass this one in the constructor as well.

**Review by carakas — CHANGES_REQUESTED** [review]

**Review by carakas — APPROVED** [review]

**Pull request merged**